src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -879,6 +911,26 @@ MultiClientRegisterWait(WaitInfo *waitInfo, TaskExecutionStatus executionStatus,
 	{
 		pollfd->events = POLLERR | POLLOUT;
 	}
+
+#else
+	connectionFileDescriptor = PQsocket(connection->pgConn);
+	if (connectionFileDescriptor > waitInfo->maxConnectionFileDescriptor)
+	{
+		waitInfo->maxConnectionFileDescriptor = connectionFileDescriptor;
+	}
+
+	if (executionStatus == TASK_STATUS_SOCKET_READ)
+	{
+		FD_SET(connectionFileDescriptor, &(waitInfo->exceptionFileDescriptorSet));
```

Comment:
Exception fd set isn't same as `POLLERR`. Based on the man pages, it is more like `POLLPRI`. So we don't need to set it here. Actually, we also don't need to set `POLLERR` for `poll`, since it is ignored in `events`.

Reply:
Addressed in https://github.com/citusdata/citus/pull/2011/commits/fd699f12534ebf78a84a0d6e77988d1a6961b0d9